src/test/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActorTest.scala

```diff
@@ -56,6 +57,63 @@ class AppTaskLauncherActorTest extends MarathonSpec with GivenWhenThen {
     Mockito.verify(taskTracker).get(app.id)
   }
 
+  test("Upgrading an app updates app definition in actor and requeries backoff") {
+    Given("an entry for an app")
+    Mockito.when(taskTracker.get(app.id)).thenReturn(Set(marathonTask))
+    val launcherRef = createLauncherRef(instances = 0)
+    rateLimiterActor.expectMsg(RateLimiterActor.GetDelay(app))
+    rateLimiterActor.reply(RateLimiterActor.DelayUpdate(app, clock.now()))
+    val counts = Await.result(launcherRef ? AppTaskLauncherActor.GetCount, 3.seconds).asInstanceOf[QueuedTaskCount]
+    assert(counts.tasksLaunchedOrRunning == 1)
+
+    When("upgrading the app")
+    val upgradedApp = app.copy(cmd = Some("new command"))
+    launcherRef ! AppTaskLauncherActor.AddTasks(upgradedApp, 1)
+
+    Then("the actor requeries the backoff delay")
+    rateLimiterActor.expectMsg(RateLimiterActor.GetDelay(upgradedApp))
+    val newDelay: Timestamp = clock.now() + 5.seconds
+    rateLimiterActor.reply(RateLimiterActor.DelayUpdate(upgradedApp, newDelay))
+    val counts2 = Await.result(launcherRef ? AppTaskLauncherActor.GetCount, 3.seconds).asInstanceOf[QueuedTaskCount]
+    assert(counts2.backOffUntil == newDelay)
+
+    And("the actor knows the new app definition")
+    assert(counts2.app == upgradedApp)
+    assert(counts2.tasksLaunchedOrRunning == 1)
```

Comment:
I tested for `offerMatcherManager` unsubscribe now.

There are other tests that test whether the backOffUntil delay is respected.